src/test/java/uk/gov/hmcts/reform/wataskmanagementapi/controllers/TaskActionsControllerTest.java

```diff
@@ -497,6 +533,109 @@ void should_return_500_response_for_tasks_deletion() {
+        AccessControlResponse mockAccessControlResponse =
+            new AccessControlResponse(mockedUserInfo, singletonList(mockedRoleAssignment));
+        when(accessControlService.getRoles(IDAM_AUTH_TOKEN)).thenReturn(mockAccessControlResponse);
+        lenient().when(launchDarklyFeatureFlagProvider.getBooleanValue(eq(FeatureFlag.WA_COMPLETION_PROCESS_UPDATE),
+                                                                       any(), anyString())).thenReturn(false);
+        CompleteTaskRequest request = new CompleteTaskRequest(null);
+
+        ResponseEntity response = taskActionsController.completeTask(
+            IDAM_AUTH_TOKEN,
+            SERVICE_AUTHORIZATION_TOKEN,
+            taskId,
+            completionProcess,
+            request
+        );
+
+        assertNotNull(response);
+        assertEquals(HttpStatus.NO_CONTENT, response.getStatusCode());
+        verify(taskManagementService, times(1)).completeTask(
+            taskId,
+            mockAccessControlResponse,
+            requestParamMap
+        );
+
+    }
+
+    @CsvSource(value = {
+        "EXUI_USER_COMPLETION",
+        "EXUI_CASE-EVENT_COMPLETION"
+    }, nullValues = "null")
+    @ParameterizedTest(name = "should complete task with valid termination process when flag enabled")
+    void should_call_complete_task_with_valid_completion_process_in_map_when_flag_enabled(String completionProcess) {
+        AccessControlResponse mockAccessControlResponse =
+            new AccessControlResponse(mockedUserInfo, singletonList(mockedRoleAssignment));
+        when(accessControlService.getRoles(IDAM_AUTH_TOKEN)).thenReturn(mockAccessControlResponse);
+
+
+        lenient().when(launchDarklyFeatureFlagProvider.getBooleanValue(eq(FeatureFlag.WA_COMPLETION_PROCESS_UPDATE),
+                                                                       any(), anyString())).thenReturn(true);
+        CompleteTaskRequest request = new CompleteTaskRequest(null);
+        doReturn(Optional.of(completionProcess)).when(completionProcessValidator)
+            .validate(anyString(), anyString(), anyBoolean());
+
+        requestParamMap = Map.of(
+            COMPLETION_PROCESS, completionProcess
+        );
+        ResponseEntity<Void> response = taskActionsController.completeTask(
+            IDAM_AUTH_TOKEN,
+            SERVICE_AUTHORIZATION_TOKEN,
+            taskId,
+            completionProcess,
+            request
+        );
+
+        assertNotNull(response);
+        assertEquals(HttpStatus.NO_CONTENT, response.getStatusCode());
+        verify(taskManagementService, times(1)).completeTask(
+            taskId,
+            mockAccessControlResponse,
+            requestParamMap
+        );
+    }
+
+    @CsvSource(value = {
+        "null",
+        "INVALID_COMPLETION_PROCESS",
+        "''"
+    }, nullValues = "null")
+    @ParameterizedTest(name = "should complete task with valid termination process when flag enabled")
```

Comment:
maybe this desc should be updated since it's about invalid parameters being passed?